pkg/pipeline/extract/conntrack/conntrack_test.go

```diff
@@ -482,14 +491,161 @@ func TestUpdateConn_Unidirectional(t *testing.T) {
 		},
 	}
 
-	for _, test := range table {
+	for _, tt := range table {
 		var prevTime time.Time
-		t.Run(test.name, func(t *testing.T) {
-			require.Less(t, prevTime, test.time)
-			prevTime = test.time
-			clk.Set(test.time)
-			actual := ct.Extract(test.inputFlowLogs)
-			require.Equal(t, test.expected, actual)
+		t.Run(tt.name, func(t *testing.T) {
+			require.Less(t, prevTime, tt.time)
+			prevTime = tt.time
+			clk.Set(tt.time)
+			actual := ct.Extract(tt.inputFlowLogs)
+			require.Equal(t, tt.expected, actual)
+		})
+	}
+}
+
+// TestIsFirst_LongConnection tests the IsFirst works right in long connections that have multiple updateConnection records.
+// In the following test, there should be 2 update connection records and 1 endConnection. Only the first updateConnection record has isFirst set to true.
+func TestIsFirst_LongConnection(t *testing.T) {
+	test.ResetPromRegistry()
+	clk := clock.NewMock()
+	updateConnectionInterval := 10 * time.Second
+	endConnectionTimeout := 30 * time.Second
+	conf := buildMockConnTrackConfig(false, []string{"updateConnection", "endConnection"}, updateConnectionInterval, endConnectionTimeout)
+	ct, err := NewConnectionTrack(opMetrics, *conf, clk)
+	require.NoError(t, err)
+
+	ipA := "10.0.0.1"
+	ipB := "10.0.0.2"
+	portA := 9001
+	portB := 9002
+	protocol := 6
+	hashIdAB := "705baa5149302fa1"
+	flAB1 := newMockFlowLog(ipA, portA, ipB, portB, protocol, 111, 11)
+	startTime := clk.Now()
+	table := []struct {
+		name          string
+		time          time.Time
+		inputFlowLogs []config.GenericMap
+		expected      []config.GenericMap
+	}{
+		{
+			"start: flow AB",
+			startTime.Add(0 * time.Second),
+			[]config.GenericMap{flAB1},
+			nil,
+		},
+		{
+			"9s: no update report",
+			startTime.Add(9 * time.Second),
+			nil,
+			nil,
+		},
+		{
+			"11s: update report AB (with isFirst=true)",
+			startTime.Add(11 * time.Second),
+			nil,
+			[]config.GenericMap{
+				newMockRecordUpdateConn(ipA, portA, ipB, portB, protocol, 111, 11, 1).withHash(hashIdAB).markFirst().get(),
+			},
+		},
+		{
+			"20s: no update report",
+			startTime.Add(20 * time.Second),
```

Comment:
Perhaps this should be 19 or less. 20 seems too borderline and it might just get an additional update report.

Reply:
Thanks for taking a close look at the tests!
Since the update report was at time `11s`, the next update should be at time `21s`. So `21s` is the borderline. That's why I check a second before (at `20s`) that there are no updates and a second after (at `22s`) that there is an update.